### PR TITLE
Show reactions to the user, rather than user created reactions to other users

### LINKED ISSUE #15242: Cannot reach the /admin page for certain users
**Describe the bug**

I'm unable to reach the admin page (/admin/users/###/edit) for my DEV account and at least [one other user's account](https://github.com/forem/pii/issues/93).

**To Reproduce**

1. Ensure you have admin privileges on DEV.
2. Go to my admin page 'https://dev.to/admin/users/38578/edit'
3. See spinning robot

**Expected behavior**

I'd expect to be able to navigate to any user's admin page.

**Additional context**

I need to be able to navigate to the admin page to via and edit user roles.

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

When we show the reports on the user edit page in admin, we want to know about reactions to the user, rather than user created reactions (similar to the way we only show reactions to the user's articles or their comments, and not reactions they made for other user's articles or comments).

I think neither the callbacks (`has_many :reactions, dependent: :destroy`) nor the `Users::DeleteActivity` class handle removing reactions _to_ the user object when we remove it - if the user's showing this error are moderators (able to add vomit reactions to users that are subsequently deleted) this makes sense around the surrounding context.

## Related Tickets & Documents

Fixes #15242 
https://github.com/forem/forem/issues/15245 seems to be the underlying issue with the data (which is not corrected here).

https://app.honeybadger.io/projects/66984/faults/78734892/

## QA Instructions, Screenshots, Recordings

Create a `User` vomit reaction by a user

```ruby
Reaction.create(
  category: 'vomit', 
  reactable: User.find(@reacted_user_id), 
  user: User.find(@reacting_user_id
)
```

Delete the reactions reacted user (the targeted user)

```ruby
User.find(@reacted_user_id).destroy

# or Users::Delete.new(User.find(@reacted_user_id)).call
```

Edit the creating/reacting user in admin and observe the changed behavior

on main - replicating the error reported:
Screenshot from 2021-11-01 12-01-45

this branch - filtering out the reaction on a deleted user:

Screenshot from 2021-11-01 12-02-25

### UI accessibility concerns?

None

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: bugfix
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] This change does not need to be communicated, and this is why not: bugfix

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -272,7 +272,7 @@ def set_related_reactions
       @related_vomit_reactions =
         Reaction.where(reactable_type: "Comment", reactable_id: user_comment_ids, category: "vomit")
           .or(Reaction.where(reactable_type: "Article", reactable_id: user_article_ids, category: "vomit"))
-          .or(Reaction.where(reactable_type: "User", user_id: @user.id, category: "vomit"))
+          .or(Reaction.where(reactable_type: "User", reactable_id: @user.id, category: "vomit"))
```
Comment (djuber) [inline app/controllers/admin/users_controller.rb:275]: rather than reactions by the user, show reactions to the user (by others, a report).

**Comment by djuber** [comment]
it looks like there's an equivalent error (with the same data present) on /admin/moderation/privileged_reactions fetching the username of a deleted user when the user reaction is still present

**Review by juliannatetreault — APPROVED** [review]
Thank you for fixing this! 🥇

**Review by Zhao-Andy — APPROVED** [review]
Ah nice catch! LGTM!

**Pull request merged**